src/js/mutations.ts

```diff
@@ -0,0 +1,143 @@
+import { useMutation } from 'react-query';
+import { useManagementClient } from '../react/ManagementProvider';
+import { CreateAccountRequest } from '../types/account';
+import {
+  useAssumeRoleQuery,
+  useS3Client,
+} from '../react/next-architecture/ui/S3ClientProvider';
+import { setRoleArnStored } from '../react/utils/localStorage';
+import { useIAMClient } from '../react/IAMProvider';
+import { TagSetItem } from '../types/s3';
+import { S3 } from 'aws-sdk';
+
+const useCreateEndpointMutation = () => {
+  const managementClient = useManagementClient();
+  if (managementClient) {
+    return useMutation({
+      mutationFn: ({
+        hostname,
+        locationName,
+        instanceId,
+        isBuiltin,
+      }: {
+        hostname: string;
+        locationName: string;
+        instanceId: string;
+        isBuiltin?: boolean;
+      }) => {
+        const params = {
+          uuid: instanceId,
+          endpoint: {
+            hostname,
+            locationName,
+            isBuiltin,
+          },
+        };
+        return managementClient.createConfigurationOverlayEndpoint(
+          params.endpoint,
+          params.uuid,
+        );
+      },
+    });
+  }
+};
+
+const useCreateAccountMutation = () => {
+  const managementClient = useManagementClient();
+  if (managementClient) {
+    return useMutation({
+      mutationFn: ({
+        user,
+        instanceId,
+      }: {
+        user: CreateAccountRequest;
+        instanceId: string;
+      }) => {
+        const params = {
+          uuid: instanceId,
+          user: { ...user, userName: user.Name },
+        };
+        return managementClient.createConfigurationOverlayUser(
+          params.user,
+          params.uuid,
+        );
+      },
+    });
+  }
+};
+
+// const roleArn = `arn:aws:iam::${accountId}:role/storage-manager-role`;
+const useAssumeRoleMutation = () => {
+  const { getQuery } = useAssumeRoleQuery();
+  return useMutation({
+    mutationFn: (roleArn: string) => getQuery(roleArn).queryFn(),
+    onSuccess(_, variables) {
+      setRoleArnStored(variables);
+    },
+  });
```

Comment:
I would useSetAssumeRole instead and return setRole among with it's result, also I would make setRole return the mutation object (without the mutate method, which would be replaced by the setRole one).